### ISSUE TITLE
sql: unaccent function

### ISSUE THREAD

**Issue by verglor** [cockroachdb/cockroach#30540]
**FEATURE REQUEST**

Please implement PostgreSQL's [unaccent() function](https://www.postgresql.org/docs/current/static/unaccent.html).

**Comment by knz** [comment]
Note: This feature is conditional on supporting PostgreSQL's full text search features, a much larger project than just a SQL built-in function. 

See #7821.

**Comment by bdarnell** [comment]
As a workaround for some use cases, the magic undocumented collations (https://github.com/cockroachdb/docs/issues/1471) can be used instead. A "level1" collation performs accent-insensitive comparisons so a unique constraint on a column of this type will perform accent-insensitive duplicate checks, for example:

```
root@127.0.0.1:58318/defaultdb> create table t (s string collate en_u_ks_level1 primary key);
CREATE TABLE

Time: 4.625ms

root@127.0.0.1:58318/defaultdb> insert into t values ('hotel'::string collate en_u_ks_level1);
INSERT 1

Time: 3.332ms

root@127.0.0.1:58318/defaultdb> insert into t values ('hôtel'::string collate en_u_ks_level1);
pq: duplicate key value (s)=('hotel' COLLATE en_u_ks_level1) violates unique constraint "primary"
```

**Comment by jordanlewis** [comment]
Closed by #54628. Thanks @mknycha!
